### PR TITLE
RavenDB - 16499 Documents Compression - All Collections

### PR DESCRIPTION
**Pull request by lastav5**
https://issues.hibernatingrhinos.com/issue/RavenDB-16499

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Raven.Client/ServerWide/DatabaseRecord.cs [hunk 1]
```diff
@@ -431,15 +431,17 @@ public enum DeletionInProgressStatus
...
         public string[] Collections { get; set; }
+        public bool CompressAllCollections { get; set; }
         public bool CompressRevisions { get; set; }
 
         public DocumentsCompressionConfiguration()
         {
         }
 
-        public DocumentsCompressionConfiguration(bool compressRevisions, params string[] collections)
+        public DocumentsCompressionConfiguration(bool compressRevisions, bool compressAllCollections, params string[] collections)
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/DatabaseRecord.cs:447]: This breaks source and binary compatibility. You need to create a separate ctor.

File: src/Raven.Client/ServerWide/Operations/DocumentCompressionConfigurationResult.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Raven.Client.ServerWide.Operations
+{
+    class DocumentCompressionConfigurationResult
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/DocumentCompressionConfigurationResult.cs]: write explicitly the modifier

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+using Raven.Client.Documents.Operations;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Json.Serialization;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.ServerWide.Operations
+{
+    class UpdateDocumentsCompressionConfigurationOperation : IMaintenanceOperation<DocumentCompressionConfigurationResult>
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: This is implicitly set to `internal`, but we want this to be `public` ?

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        private readonly DocumentsCompressionConfiguration _documentsCompressionConfiguration;
+
+        public UpdateDocumentsCompressionConfigurationOperation(DocumentsCompressionConfiguration documentsCompressionConfiguration)
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: rename the paramter to `configuration`

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        public RavenCommand<DocumentCompressionConfigurationResult> GetCommand(DocumentConventions conventions, JsonOperationContext context)
+        {
+            return new UpdateDocumentCompressionConfigurationCommand(_documentsCompressionConfiguration);
+        }
+
+        private class UpdateDocumentCompressionConfigurationCommand : RavenCommand<DocumentCompressionConfigurationResult>, IRaftCommand
+        {
+            private readonly DocumentsCompressionConfiguration _documentsCompressionConfiguration;
+
+            public UpdateDocumentCompressionConfigurationCommand(DocumentsCompressionConfiguration documentsCompressionConfiguration)
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: let's simplify the parameter name, rename it to `configuration`

File: src/Raven.Client/ServerWide/DatabaseRecord.cs (hunk 1 above)
Comment (lastav5) [reply src/Raven.Client/ServerWide/DatabaseRecord.cs:447]: Done

File: src/Raven.Client/ServerWide/Operations/DocumentCompressionConfigurationResult.cs (hunk 2 above)
Comment (lastav5) [reply src/Raven.Client/ServerWide/Operations/DocumentCompressionConfigurationResult.cs]: Done

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs (hunk 3 above)
Comment (lastav5) [reply src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: Done

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs (hunk 4 above)
Comment (lastav5) [reply src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: Done

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs (hunk 5 above)
Comment (lastav5) [reply src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: Done

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+using System.Net.Http;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Json.Serialization;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.ServerWide.Operations
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: Let's not put everything in one bag and create a dedicated namespace for DocumentsCompression
e.g. move it to `Raven.Client.ServerWide.Operations.DocumentsCompression`

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs (hunk 6 above)
Comment (lastav5) [reply src/Raven.Client/ServerWide/Operations/UpdateDocumentsCompressionConfigurationOperation.cs]: Done

**Comment by ppekrol** [comment]
@ml054 Could you take care of the Studio part?

**Comment by ml054** [comment]
created issue: RavenDB-16910

**Pull request merged**